UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -65,18 +67,28 @@ public async Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMe
                 List<FulfilmentDataResponse> fulfilmentDataResponse = new List<FulfilmentDataResponse>();
                 object sync = new object();
                 int fileShareServiceSearchQueryCount = 0;
+                var cTs = new CancellationTokenSource();
+                CancellationToken cToken = cTs.Token;
+
                 var tasks = productsList.Select(async item =>
                 {
-                    fulfilmentDataResponse = await QueryAndDownloadFileShareServiceFiles(message, item, exchangeSetRootPath);
+                    fulfilmentDataResponse = await QueryAndDownloadFileShareServiceFiles(message, item, exchangeSetRootPath, cTs, cToken);
                     int queryCount = fulfilmentDataResponse.Count > 0 ? fulfilmentDataResponse.FirstOrDefault().FileShareServiceSearchQueryCount : 0;
                     lock (sync)
                     {
                         fileShareServiceSearchQueryCount += queryCount;
                     }
+                    if (cToken.IsCancellationRequested)
+                    {
+                        cTs.Cancel();                      
+                        logger.LogError(EventIds.CancellationTokenEvent.ToEventId(), "Operation is cancelled in QueryAndDownloadFileShareServiceFiles with {cancellationTokenSource.Token} at Time:{DateTime.UtcNow} and batchId:{message.BatchId} and CorrelationId:{message.CorrelationId}", JsonConvert.SerializeObject(cTs.Token), DateTime.UtcNow, message.BatchId, message.CorrelationId);
```

Comment:
Kindly add "Operation is cancelled as IsCancellationRequested flag is true" to maintain consistency across cancellation logs